libs/@local/codegen/src/typescript.rs

```diff
@@ -148,15 +150,86 @@ impl<'a, 'c> TypeScriptGenerator<'a, 'c> {
         }
     }
 
-    fn visit_type_definition(&self, definition: &TypeDefinition) -> ast::Declaration<'a> {
-        if self.should_export_as_interface(&definition.r#type) {
+    fn visit_type_definition(&mut self, definition: &TypeDefinition) -> ast::Declaration<'a> {
+        if !definition.branded && self.should_export_as_interface(&definition.r#type) {
             self.generate_interface(definition)
         } else {
+            let mut r#type = self.visit_type(&definition.r#type);
+
+            if definition.branded {
+                if !self.has_branded_types {
+                    self.has_branded_types = true;
+                    self.add_import_declaration("@local/advanced-types/brand", ["Brand"]);
```

Comment:
couldn't this be more robustly solved through HashMap+HashSet instead? I could imagine that there are quite a few imports in the future and using booleans seems counter productive.

Reply:
I thought the same and I likely will change it to be a map instead, but not in this PR :)